sale_stock_available_to_promise_release/models/sale_order_line.py

```diff
@@ -1,13 +1,73 @@
 # Copyright 2019 Camptocamp (https://www.camptocamp.com)
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
-from odoo import models
+from odoo import api, fields, models
+from odoo.tools import float_compare, float_is_zero
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    def _prepare_procurement_values(self, group_id=False):
-        values = super()._prepare_procurement_values(group_id)
-        values["date_priority"] = self.order_id.date_order
-        return values
+    availability_status = fields.Selection(
+        selection=[
+            ("mto", "On order"),
+            ("full", "Fully Available"),
+            ("partial", "Partially Available"),
+            ("restock", "Restock ordered"),
+            ("no", "Not available"),
+        ],
+        compute="_compute_availability_status",
+    )
+    expected_availability_date = fields.Datetime(compute="_compute_availability_status")
+    available_qty = fields.Float(
+        digits="Product Unit of Measure", compute="_compute_availability_status"
+    )
+
+    @api.depends(
+        "is_mto",
+        "move_ids.ordered_available_to_promise_uom_qty",
+        "product_id",
+        "product_uom_qty",
+    )
+    def _compute_availability_status(self):
+        for record in self:
+            # Fallback values
+            availability_status = "no"
+            expected_availability_date = False
+            available_qty = sum(
+                record.mapped("move_ids.ordered_available_to_promise_uom_qty")
+            )
+            # required values
+            product = record.product_id
+            rounding = product.uom_id.rounding
+            is_fully_available = (
+                float_compare(
+                    available_qty, record.product_uom_qty, precision_rounding=rounding
+                )
+                >= 0
+            )
+            is_partially_available = (
```

Comment:
If it is fully available it would be better to not do this computation as well no ? Improved  performance